tests/ConfigTest.php

```diff
@@ -25,6 +26,41 @@
  */
 final class ConfigTest extends \PHPUnit_Framework_TestCase
 {
+    public function testConfigRulesUsingJson()
```

Comment:
nice cases :)
when you are done with the creating all tests please split up the cases in separate methods, pref. using dataproviders where handy.